src/read/read_options.rs

```diff
@@ -112,8 +137,94 @@ impl ReadOptions {
         self
     }
 
-    /// Reads a file into a [`PDB`] structure.
-    pub fn read(&self, path: &str) -> ReadResult {
-        open_with_options(path, self)
+    /// Sets whether to only parse `ATOM` records in the model file.
+    pub fn set_only_atomic_coords(&mut self, only_atomic_coords: bool) -> &mut Self {
+        self.only_atomic_coords = only_atomic_coords;
+        self
+    }
+
+    /// Open an atomic data file, either PDB or mmCIF/PDBx, into a [`PDB`] structure.
+    /// The correct type will be determined based on the file extension.
+    ///
+    /// # Errors
+    /// Returns a `PDBError` if a `BreakingError` is found. Otherwise it returns the PDB with all errors/warnings found while parsing it.
+    ///
+    /// # Related
+    /// If you want to open a file from memory see [`ReadOptions::read_raw`].
+    pub fn read(&self, path: impl AsRef<str>) -> ReadResult {
+        if self.decompress {
+            // open a decompression stream
```

Comment:
Do I read the code correctly that the decompress branch does not auto detect the file type? Here the filename is still known so it could do the auto detect functionality.

Reply:
You are absolutely right! I was hoping users to call `self.guess_format` explicitly before calling `read`, otherwise `read` would need a mutable reference of `self` since `self.format` would be modified when guessing the format from the file name. Do you have recommendations for handling this?

Reply:
I think that having a generalised function that takes a path and returns a `Format` (or something that also included the compression when needed) would solve your problem. The `self.guess_format` then calls that and updates itself while any place can just use the result of the generalised function locally and does not need to store anything per se.